### PR TITLE
Hanan/create split

### PR DESCRIPTION
**Pull request by HananINouman**

### CODE REVIEW

File: src/ajv.ts [hunk 1]
```diff
@@ -24,7 +25,18 @@ function validDepositAmounts(data: boolean, deposits: string[]): boolean {
...
+
+const validateSplitRecipients = (
+  _: boolean,
+  data: SplitRecipient[],
+): boolean => {
+  const totalPercentage = data.reduce(
+    (acc: number, curr: SplitRecipient) => acc + curr.percentAllocation,
+    0,
+  );
+  return totalPercentage === 99;
```
Comment (LukeHackett12) [inline src/ajv.ts:38]: what if they want to pay us more 😅 ? e.g etherfi I think does a 3% split of some kind right?

File: src/constants.ts [hunk 2]
```diff
@@ -136,3 +137,23 @@ export const TERMS_AND_CONDITIONS_URL =
...
 export const TERMS_AND_CONDITIONS_HASH =
   '0xd33721644e8f3afab1495a74abe3523cec12d48b8da6cb760972492ca3f1a273';
+
+export const AVAILABLE_SPLITTER_CHAINS = [
+  FORK_MAPPING['0x00000000'],
+  FORK_MAPPING['0x01017000'],
+];
+
+export const CHAIN_CONFIGURATION = {
+  [AVAILABLE_SPLITTER_CHAINS[0]]: {
```
Comment (LukeHackett12) [inline src/constants.ts:147]: 🪆

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -32,7 +32,9 @@ jobs:
...
         working-directory: ./test/sdk-package-test
         run: yarn run build
+        env:
+          PRIVATE_KEY: ${{ secrets.PRIVATE_KEY }}
 
       - name: Run e2e tests
         working-directory: ./test/sdk-package-test
-        run: yarn run test:e2e
\ No newline at end of file
+        run: yarn run test:e2e
```
Comment (LukeHackett12) [inline .github/workflows/ci.yml:40]: probably need the env here?
```suggestion
        run: yarn run test:e2e
        env:
          PRIVATE_KEY: ${{ secrets.PRIVATE_KEY }}
```

File: test/sdk-package-test/utils.ts [hunk 4]
```diff
@@ -1,18 +1,21 @@
...
   Client,
   type ClusterLock,
 } from '@obolnetwork/obol-sdk';
+import dotenv from 'dotenv';
 
-const mnemonic = ethers.Wallet.createRandom().mnemonic?.phrase ?? '';
+dotenv.config();
 
-const privateKey = ethers.Wallet.fromPhrase(mnemonic).privateKey;
+const privateKey = process.env.PRIVATE_KEY as string; // keep it in gh secrets
```
Comment (LukeHackett12) [inline test/sdk-package-test/utils.ts:12]:
```suggestion
const privateKey = process.env.PRIVATE_KEY as string;
```

File: src/index.ts [hunk 5]
```diff
@@ -96,6 +111,100 @@ export class Client extends Base {
...
+   * @param {RewardsSplitPayload} rewardsSplitPayload - Data needed to deploy owr and splitter.
+   * @returns {Promise<SplitterReturnedType>} owr address as withdrawal address and splitter as fee recipient
+   */
+  // add the example reference
+  async createObolRewardSplit({
+    splitRecipients,
+    principalRecipient,
+    validatorsSize,
+  }: RewardsSplitPayload): Promise<SplitterReturnedType> {
+    // This method doesnt require T&C signature
```
Comment (LukeHackett12) [inline src/index.ts:125]: why shouldn't it?

File: src/index.ts [hunk 6]
```diff
@@ -96,6 +111,100 @@ export class Client extends Base {
...
+    ) {
+      throw new Error(
+        'Something isn not working as expected, check this issue with obol-sdk team',
+      );
+    }
+
+    const retroActiveFundingRecipient = {
+      account: CHAIN_CONFIGURATION[this.chainId].RETROACTIVE_FUNDING_ADDRESS,
+      percentAllocation: RETROACTIVE_FUNDING_SPLIT,
+    };
```
Comment (LukeHackett12) [inline src/index.ts:175]: could we make it that they just need to specify the percentage that the address gets, and it has to be >= 1%?

**Review by LukeHackett12 — APPROVED** [review]

File: src/index.ts [hunk 7]
```diff
@@ -96,6 +115,203 @@ export class Client extends Base {
...
+      this.signer.provider as Provider,
+    );
+
+    if (
+      !checkMulticallAddress ||
+      !checkSplitMainAddress ||
+      !checkOWRFactoryAddress
+    ) {
+      throw new Error(
+        'Something isn not working as expected, check this issue with obol-sdk team',
```
[the remark this answers is not in the document]
Comment (HananINouman) [reply src/index.ts:177]: amazing

File: src/utils.ts [hunk 8]
```diff
@@ -54,3 +55,18 @@ export const definitionFlow = (
...
+  provider: Provider,
+): Promise<string> => {
+  return await provider?.getCode(contractAddress);
+};
+
+export const isContractAvailable = async (
+  contractAddress: string,
+  provider: Provider,
+): Promise<boolean> => {
+  const code = await findDeployedBytecode(contractAddress, provider);
```
Comment (LukeHackett12) [inline src/utils.ts:70]: Only thing here, can we check that not only is it a contract but it is the specific contract we need

**Review by LukeHackett12 — APPROVED** [review]
think this looks good! some small comments, once it is merged/released might take another pass at the documentation after playing around with it

File: src/index.ts [hunk 9]
```diff
@@ -96,6 +116,203 @@ export class Client extends Base {
     }
   }
 
+  /**
+   * Deploys OWR and Splitter Proxy.
+   * @param {RewardsSplitPayload} rewardsSplitPayload - Data needed to deploy owr and splitter.
+   * @returns {Promise<ClusterValidator>} owr address as withdrawal address and splitter as fee recipient
+   */
+  // add the example reference
```
[the remark this answers is not in the document]
Comment (HananINouman) [reply src/index.ts:124]: yeah but can add it once i add the example in obol-sdk-examples

**Pull request merged**